src/cephadm/cephadm.py

```diff
@@ -699,6 +700,68 @@ def _prevalidate(self, ctx: CephadmContext) -> None:
                     )
                 )
 
+    def get_daemon_args(self) -> List[str]:
+        ctx = self.ctx
+        daemon_type = self.identity.daemon_type
+        metadata = self.components[daemon_type]
+        r = list(metadata.get('args', []))
+        # set ip and port to bind to for nodeexporter,alertmanager,prometheus
+        if daemon_type not in ['grafana', 'loki', 'promtail']:
```

Comment:
not required for this patch, but given we seem to want to set this for 3 daemon types and at this point the check is checking that it's not 3 other daemon types, this should probably get updated to 
```
if daemon_type in ['node-exporter', 'alertmanager', 'prometheus']:
```
at some point.